client/my-sites/stats/stats-chart-tabs/index.jsx

```diff
@@ -275,58 +255,47 @@ class StatModuleChartTabs extends Component {
 }
 
 const connectComponent = connect(
-	( state, { moment, period: periodObject, chartTab, queryDate } ) => {
+	( state, { period: { period }, chartTab, queryDate } ) => {
 		const siteId = getSelectedSiteId( state );
-		const { period } = periodObject;
-		const timezoneOffset = getSiteOption( state, siteId, 'gmt_offset' ) || 0;
-		const momentSiteZone = moment().utcOffset( timezoneOffset );
-		let date = rangeOfPeriod( period, momentSiteZone.locale( 'en' ) ).endOf;
-
-		let quantity = 30;
-		switch ( period ) {
-			case 'year':
-				quantity = 10;
-				break;
-		}
-		const periodDifference = moment( date ).diff( moment( queryDate ), period );
-		if ( periodDifference >= quantity ) {
-			date = moment( date )
-				.subtract( Math.floor( periodDifference / quantity ) * quantity, period )
-				.format( 'YYYY-MM-DD' );
+		if ( ! siteId ) {
+			return { siteId, data: [] };
```

Comment:
let's create something in the module scope like this; otherwise we'll be re-rendering on every render until we get a valid `siteId` even though we don't need to due to shallow comparison

```js
const noSiteState = {
	siteId: null,
	data: [],
}

…

if ( ! siteId ) {
	return noSiteState;
}
```